src/main/scala/mesosphere/marathon/state/GroupManager.scala

```diff
@@ -218,19 +218,22 @@ class GroupManager @Inject() (
       port
     }
 
-    def mergeServicePortsAndPortDefinitions(portDefinitions: Seq[PortDefinition], servicePorts: Seq[Int]) = {
-      portDefinitions.zipAll(servicePorts, AppDefinition.RandomPortDefinition, AppDefinition.RandomPortValue).map {
```

Comment:
This is inserting service ports even when `portDefinitions` is empty! So I fixed it, and then service ports weren't merging properly with container port mappings. So I changed those bits too.